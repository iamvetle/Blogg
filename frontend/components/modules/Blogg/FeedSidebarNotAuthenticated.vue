<template>
    <div>
        <UCard>
            <template #header>
                <div class="prose">
                    <h2>Create an account</h2>
                </div>
            </template>

            <template #default>
                <div class="px-4 prose">
                    <span class="block font-bold mb-2">By creating an account you can:</span>
                    <ul class="">
                        <li>Bookmark posts</li>
                        <li>Create you own posts</li>
                        <li>Follow users</li>
                        <li>Comment on posts</li>
                    </ul>
                </div>
            </template>
            <template #footer>
                <div class="space-x-6 flex justify-center">
                    <NuxtLink to="/login">
                        <UButton data-test="login_here" class="hover:shadow-md hover:bg-onPrimary text-primary bg-onPrimary ring-primary ring-1">Login</UButton>
                    </NuxtLink>
                    <NuxtLink to="/registrer">
                        <UButton data-test="register_here" class="hover:shadow-md hover:bg-primary bg-primary" >Register</UButton>
                    </NuxtLink>
                </div>
            </template>
        </UCard>
    </div>
</template>

<script setup lang="ts">

</script>

<style scoped>

</style>