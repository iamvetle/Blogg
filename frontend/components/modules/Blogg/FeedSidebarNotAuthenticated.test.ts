import FeedSidebarNotAuthenticated from './FeedSidebarNotAuthenticated.vue';
import { shallowMount } from '@vue/test-utils';
import { describe, expect, test, beforeEach, afterEach } from 'vitest';

let wrapper: any;

const factory = () => {
    return shallowMount(FeedSidebarNotAuthenticated, {
        global: {
            plugins: [],
            components: {
            },
            mocks: {},
            stubs: {
                UButton:true,
                "nuxt-link":true
            },
        },
        props: {},
        slots: {}
    })
};

describe('Testing the sidebar that is used when the web client isnt logged in/authenticated', () => {

    beforeEach(() => {
    });

    afterEach(() => {
        if (wrapper) {
            wrapper.unmount();
        }
        vi.clearAllMocks()

    });

    test('Should exist', () => {
        wrapper = factory()
        expect(wrapper.exists()).toBe(true)
    })
    
    /**
     * * Doesnt work because it is inside #footer and not just default - and I don't have the "source" for Ucard to be able to test it
     */
    // test('Should render a loggin "button" thing', () => {
    //     wrapper = factory()
    //     console.log(wrapper.html())
        

    //     expect(wrapper.find("[data-test='login_here']").exists()).toBe(true)
    // })
    // test('Should render a register "button" thing', () => {
    //     wrapper = factory()

    //     expect(wrapper.find("[data-test='register_here']").exists()).toBe(true)
    // })
    test('Should match snapshot', () => {
        wrapper = factory()
        expect(wrapper.html()).toMatchSnapshot()
    })
    
});