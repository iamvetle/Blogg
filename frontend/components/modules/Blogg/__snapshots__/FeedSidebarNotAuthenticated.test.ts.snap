// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the sidebar that is used when the web client isnt logged in/authenticated > Should match snapshot 1`] = `
"<div>
  <ucard>
    <div class=\\"px-4 prose\\"><span class=\\"block font-bold mb-2\\">By creating an account you can:</span>
      <ul class=\\"\\">
        <li>Bookmark posts</li>
        <li>Create you own posts</li>
        <li>Follow users</li>
        <li>Comment on posts</li>
      </ul>
    </div>
  </ucard>
</div>"
`;
